PolyEngine/Core/Src/EnumUtils.hpp

```diff
@@ -120,6 +126,13 @@ namespace Poly {
 		const T& operator[](size_t idx) const { HEAVY_ASSERTE(idx < GetSize(), "Index out of bounds!"); return Data[idx]; }
 
 		//------------------------------------------------------------------------------
+		void PopulateFromInitializerList(const std::initializer_list<std::pair<E, T>>& list)
+		{
+			for (const std::pair<E, T>& val : list)
+				Data[static_cast<int>(val.first)] = val.second;
+		}
+
+		//------------------------------------------------------------------------------
 		void PopulateFromInitializerList(const std::initializer_list<T>& list)
```

Comment:
Let's remove creation from `std::initializer_list<T>` alltogether and either force to declare enum names by providing both the enum value and enum name or refactor enum names declaration to create `std::initializer_list<E,T>` on the fly. Preferably both.